File: backend/schemas/preferences.py
from pydantic import BaseModel, Field
from typing import Dict, Optional, List
from datetime import time


class AccessibilityPreferences(BaseModel):
    high_contrast: bool = Field(default=False)
    large_text: bool = Field(default=False)
    reduced_motion: bool = Field(default=False)
    text_scale: int = Field(default=100, ge=50, le=200)


class NotificationPreferences(BaseModel):
    email_notifications: bool = Field(default=True)
    push_notifications: bool = Field(default=False)
    quiet_hours_start: time = Field(default=time(22, 0))  # 10 PM
    quiet_hours_end: time = Field(default=time(7, 0))     # 7 AM


class AIAssistancePreferences(BaseModel):
    assistance_level: str = Field(default="balanced", pattern="^(low|balanced|high)$")
    auto_categorize: bool = Field(default=True)
    stress_monitoring: bool = Field(default=True)


class PreferencesUpdate(BaseModel):
    accessibility: Optional[AccessibilityPreferences] = None
    notifications: Optional[NotificationPreferences] = None
    ai_assistance: Optional[AIAssistancePreferences] = None


class PreferencesResponse(BaseModel):
    accessibility: AccessibilityPreferences
    notifications: NotificationPreferences
    ai_assistance: AIAssistancePreferences

File: src/types/preferences.ts
export interface SensoryPreferences {
  fontScale: number;
  contrast: 'normal' | 'high';
  motion: 'normal' | 'reduced';
  textSpacing: 'normal' | 'increased' | 'maximum';
  colorMode: 'default' | 'deuteranopia' | 'protanopia' | 'tritanopia';
  lineHeight: number;
  fontFamily: string;
  readingSpeed: 'slow' | 'normal' | 'fast';
}

export interface AccessibilityPreferences {
  high_contrast: boolean;
  large_text: boolean;
  reduced_motion: boolean;
  text_scale: number;
}

export interface NotificationPreferences {
  email: boolean;
  push: boolean;
  quiet_hours: {
    start: string;
    end: string;
  };
}

export interface AIAssistancePreferences {
  level: 'minimal' | 'balanced' | 'high';
  auto_categorize: boolean;
  stress_monitoring: boolean;
}

export interface PreferencesUpdate {
  accessibility?: Partial<AccessibilityPreferences>;
  notifications?: Partial<NotificationPreferences>;
  ai_assistance?: Partial<AIAssistancePreferences>;
  theme?: string;
}

export interface PreferencesResponse {
  accessibility: AccessibilityPreferences;
  notifications: NotificationPreferences;
  ai_assistance: AIAssistancePreferences;
  theme: string;
} 